### ISSUE TITLE
Arguments Error in Search.rb when initializing Client Object

### ISSUE THREAD

**Issue by isFakeAccount** [games-directory/api-psn#2]
https://github.com/games-directory/api-psn/blob/090ced99afeaf9ec371999bf05f370e2f90715c5/lib/play_station_network_api/search.rb#L13

The following line will give this error

> C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/psn-api-2.0/lib/play_station_network_api/client.rb:12:in `initialize': wrong number of arguments (given 0, expected 1) (ArgumentError)

You either need to pass the `refresh_token` along with `query` or maybe just create a new method post in search and call super in. Something like this. 

Calling 
```
request = self.post('/search/v1/universalSearch',
```
Super Method
```
def post(url, options = {})
   super
end
```

Although during my testing it just returned `{:total_results=>nil, :results=>nil}` every time. I am not sure if I am doing something wrong or perhaps the endpoints are not working. 

This is how I am calling the function
```
client = PlayStationNetworkAPI::Client.new(refresh_token)
search = client.search
puts search.query(query="VaultTec_Trading")
``` 

I tried following the decompiled sony apk. But it is definitely hard to follow. I am looking for an endpoint that would allow me to get Account ID from PSN ID.

**Comment by rbhowsden** [comment]
`
headers = {'Authorization': 'Bearer ' + output['access_token']}

requests.get("https://us-prof.np.community.playstation.net/userProfile/v1/users/Hakoom/profile2?fields=accountId%2ConlineId", headers=headers)`

I'm not going to be much help on the Ruby version - I spent a lot of time today trying to figure out how to get accountId from onlineId...  I'm guessing though with the new API for Sony, you won't be able to find accountIds from onlineIds.

**Comment by pacMakaveli** [comment]
@isFakeAccount Indeed, there was an issue with that class. I must've forgotten to update the code when I extracted the main logic into the Client class. I've fixed the issue with #

Couple of things. This still doesn't work because I need to find a way to pass Java JSON with HTTParty ( the http library )
However, if you're doing JAVA already, here's the request.
```
curl --location --request POST 'https://m.np.playstation.net/api/search/v1/universalSearch' \
--header 'Content-Type: application/json' \
--header 'Accept-Language: en-US' \
--header 'Authorization: Bearer {{access_token}}' \
--data-raw '{ 
    "domainRequests": [
        {
            "domain": "ConceptGameMobileApp",
            "pagination": {
                "pageSize": 50,
                "offset": 0
            }
        }
    ],
    "age": "999",
    "countryCode": "US",
    "searchTerm": "Rocket League",
    "languageCode": "en"
}'
```
As for your main issue, I don't think you can actually get an account ID through this endpoint.
If it is indeed possible, then the searchDomain would have to be different from what I can tell. Currently this is hardcoded to `ConceptGameMobileApp` which is something I got from the app. For users that could be changed to be something `ProfileMobileApp` (?) I'm spitballing here by the way.

I haven't found anything in the app to be honest, but in all fairness I haven't looked very hard as my goal with this gem was to be able to get a logged in user's games and details.

However, to get an account id, there's a couple of ways.

1. Through dms.api
https://dms.api.playstation.com/api/v1/devices/accounts/me
This only works for the current user though, which is you. As far as I can tell, DMS seems to be a restricted API that only works for logged in users.
But in my case this is what I needed because on games.directory users have to authenticate anyway.

2. Through the old endpoint as @rbhowsden suggested
https://github.com/games-directory/api-psn/blob/v1.0/lib/play_station_network_api/user.rb#L16
```
curl --location --request GET 'https://gb-prof.np.community.playstation.net/userProfile/v1/users/Hakoom/profile2?fields=npId,onlineId,accountId,currentOnlineId' \
--header 'Content-Type: application/json' \
--header 'Accept-Language: en-US' \
--header 'Authorization: Bearer {{access_token}}'
```

**Comment by pacMakaveli** [comment]
So, you got me very curious and I had to dig a bit more into the apk. Turns out you were onto something.

```
BLENDED: 'Blended',
CONCEPT_GAME: 'ConceptGame',
CONCEPT_GAME_ALL: 'ConceptGameAll',
CONCEPT_GAME_MOBILE_APP: 'ConceptGameMobileApp',
CONCEPT_GAME_PS_NOW: 'ConceptGamePsNow',
CONCEPT_GAME_EXCLUDING_PS_NOW: 'ConceptGameExcludingPsNow',
CONCEPT_GAME_ADDONS: 'ConceptGameAddons',
CONCEPT_APPLICATION_GAME: 'ConceptApplicationGame',
CONCEPT_APPLICATION_VIDEO: 'ConceptApplicationVideo',
CONCEPT_OTHER_NON_GAME: 'ConceptOtherNonGame',
VIDEO_CATALOG: 'VideoCatalog',
VIDEO_CATALOG_MOVIE: 'VideoCatalogMovie',
VIDEO_CATALOG_TV_SERIES: 'VideoCatalogTVSeries',
SOCIAL_ALL_ACCOUNTS: 'SocialAllAccounts',
SOCIAL_FRIENDS: 'SocialFriends',
SOCIAL_FRIENDS_OF_FRIENDS: 'SocialFriendsOfFriends',
SOCIAL_FRIENDS_AND_FRIENDS_OF_FRIENDS: 'SocialFriendsAndFriendsOfFriends',
SOCIAL_UNRELATED_AND_FRIENDS_OF_FRIENDS: 'SocialUnrelatedAndFriendsOfFriends',
SOCIAL_UNRELATED: 'SocialUnrelated'
```

These are possible values for the `searchDomains` , so far.
Couple of things I noticed, not sure that they are important, but nevertheless.

1. If you try to search for yourself as the logged in user, it won't return anything.
2. You can combine domains.

```
curl --location --request POST 'https://m.np.playstation.net/api/search/v1/universalSearch' \
--header 'Content-Type: application/json' \
--header 'Accept-Language: en-US' \
--header 'Authorization: Bearer {{access_token}}' \
--data-raw '{ 
    "domainRequests": [
        {
            "domain": "SocialAllAccounts",
            "pagination": {
                "pageSize": 50,
                "offset": 0
            }
        },
        {
            "domain": "ConceptGameMobileApp",
            "pagination": {
                "pageSize": 50,
                "offset": 0
            }
        }
    ],
    "age": "999",
    "countryCode": "GB",
    "searchTerm": "Rocket League",
    "languageCode": "en"
}'
```

Anyway, thanks for this issue! Don't think I would've bothered otherwise. Some interesting Search Domains in there, can't wait to dig some more.

@isFakeAccount @rbhowsden
